### PR TITLE
[Merged by Bors] - chore (Data.Nat.Cast.Field): split into unbundled and bundled results 

### PR DESCRIPTION
**Pull request by mattrobball**
We split off the results depending on `LinearOrderedSemifield` into a separate file.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
Needs a `lake exe mk_all` (and one long line being fixed).

File: Mathlib/Data/Nat/Cast/Field.lean [hunk 1]
```diff
@@ -3,9 +3,13 @@ Copyright (c) 2014 Mario Carneiro. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Mario Carneiro, Yaël Dillies, Patrick Stevens
 -/
-import Mathlib.Algebra.Order.Field.Basic
-import Mathlib.Data.Nat.Cast.Order.Basic
+import Mathlib.Algebra.CharZero.Defs
+-- import Mathlib.Algebra.Order.Field.Basic
+-- import Mathlib.Data.Nat.Cast.Order.Basic
```
Comment (grunweg) [inline Mathlib/Data/Nat/Cast/Field.lean:8]:
```suggestion

**Comment by grunweg** [comment]
I took the liberty to push a simple fix for the archive build failure - I hope you don't mind. (If you do, I'm sorry and will not do so again!)

**Review by grunweg — COMMENTED** [review]
I cannot vouch for the change in `Mathlib/Algebra/Order/Monoid/Unbundled/Basic.lean`; the rest looks good. Thank you for doing this!
maintainer merge

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**